.github/workflows/check-semgrep-pro.jsonnet

```diff
@@ -7,53 +7,60 @@ local gha = import 'libs/gha.libsonnet';
 local actions = import 'libs/actions.libsonnet';
 local semgrep = import 'libs/semgrep.libsonnet';
 
+// exported for other workflows
```

Comment:
Why not just porting the tests that we want to run to the Pro repo? We need to do that in preparation for the monorepo.

Reply:
I did that in the past, but I'm not a fan of duplicating the pytest test infrastructure. Since `--pro-intrafile` will be default, I want all the semgrep ci tests to be run using `--pro-intrafile`. If someone adds a `semgrep ci` test, I don't want them to have to go to a different repo to make it use the correct binary. @kopecs also had some tests he wanted to add that made more sense in the semgrep repo.

> This will not work when the PR requires another PR in Pro
That's true, but when that happens CI is red anyway usually

Reply:
> How much do we need to duplicate?

I'm not quite sure, actually. Is your vision that the pytests are all entirely duplicated on the semgrep-proprietary side? And that we add new pytests for pro in the semgrep repo? I think that wouldn't be too much effort, if it's just a mirror, and I could be convinced to support that. My main concern is that sometimes you might not want to add a pytest on the semgrep side for various reasons (e.g. testing a new rule format SR wants to keep quiet)

Reply:
If we want to do things well, I think there is no choice than to run for each Pro PR all the checks we run for OSS PRs. (This is actually regardless of having a monorepo or not having it.) If we disagree on this point we should have a video call Um, you and I, because I would like to understand your point of view, I could be wrong.

Reply:
> run for each Pro PR all the checks we run for OSS PRs

I agree with this direction.

Regarding this PR in particular, what do you think is a good way to test changes like [making pro-intrafile turned on by default](https://github.com/semgrep/semgrep/pull/9872)?

Should it be a manual trigger from the pro repo instead, where the dev can insert a PR to test?